patroni/ha.py

```diff
@@ -877,11 +905,21 @@ def _is_healthiest_node(self, members: Collection[Member], check_replication_lag
         members = [m for m in members if m.name != self.state_handler.name and not m.nofailover and m.api_url]
 
         if members:
-            for st in self.fetch_nodes_statuses(members):
+            for st, priority in self.fetch_nodes_statuses_and_priorities(members):
                 if st.failover_limitation() is None:
                     if st.in_recovery is False:
                         logger.warning('Primary (%s) is still alive', st.member.name)
                         return False
+                    if self.is_lagging(st.wal_position):
+                        # The other node appears to have intolerable lag, move on (regardless of priority)
+                        continue
```

Comment:
There is no point in having it here. This method is checking if there are members that are ahead of our node in replication and we don't care if these nodes are "lagging" in term of `maximum_lag_on_failover`.

Reply:
This makes sense. The reason I had this here was because of the following concern:

Consider a situation where we have a three node system. Two of the nodes, call them A and B, have `failover_priority: 2` and the third, call it C, has `failover_priority: 1` (I imagine this will be a pretty typical configuration for people who chose to use the priority setting).

Say that leadership is originally on A but it goes down. Assume as well that something is wrong with B and it is outside of the `maximum_lag_on_failover`, but is still up. This means that B sees itself as lagging and refuses to race for leadership.

If we don't have this check, then C will see that B is up and has higher priority and so it will also not race for leadership. Thus, we'll end up in a scenario where no one goes for leadership even though both B and C have positive `failover_priority` (which we want to mean that 

Is there something I'm missing about why this either isn't a concern, or why this check won't help prevent it?

Reply:
We need to check priority only when the other node has the same wal_position as ours.
I.e., add an additional check (`if my_wal_position == st.wal_position`) to https://github.com/zalando/patroni/pull/2780/files#diff-b6a3a0ef64783a6d32feb908d74a3a009e9f25008b8ca5313c9c9747e3b74a63R897